improver/blending/blend_across_adjacent_points.py

```diff
@@ -157,9 +156,7 @@ def process(self, cube):
         weights = self.WeightsPlugin(cube, self.coord, self.central_point)
         blended_cube = self.BlendingPlugin(cube, weights)
 
-        # With one threshold dimension (such as for low cloud), the threshold
-        # axis is demoted to a scalar co-ordinate by BlendingPlugin. This line
-        # promotes threshold to match the dimensions of central_point_cube.
-        blended_cube = check_cube_coordinates(central_point_cube, blended_cube)
```

Comment:
Does this case no longer occur?  Is the assumption now that a cube with a one-value dimension coordinate would not have it sliced down, as it would not meet the size criteria for slicing?

Reply:
No, I think this might be still needed. Do we have any test for it?

Reply:
No; all of the acceptance test input data have multiple thresholds or percentiles.

Reply:
I think unit test would do.

(BTW, this is another reason why `if bigger then size` conditions are bad - it's hard to test them)

Reply:
I'm not sure it's needed there though, I think all the metadata is just copied from the central cube anyway?

Reply:
I will check whether this is needed after the acceptance tests pass.

Reply:
Okay, I've added a test for TriangularWeightedBlending that promotes a dimension. The upshot of it is that this statement **is** needed, but not there. The acceptance test that @fionaRust pointed out is failing, required that statement in the WeightAcrossWholeDimension  plugin, by putting it there, we can pass a cubeList containing cubes with a single-coordinate-dimension, that would ordinarily get squeezed,  into TriangularWeightedBlend and receive the correct number of dimensions in the blended cube. 

Have a check and see what you think once I push it, but I think it solves 2 problems by putting it where it is.

Reply:
I find this `check_cube_coordinates` routine quite confusing. I think it would be much clearer to have `promote_coordinates(cube, ref_cube) -> None`, i.e. an in-place routine that modifies the first argument (common practice) and doesn't return any result.

Reply:
If you are going to change the check_cube_coordinates function please do it in a separate PR. It is used all over the place

Reply:
> If you are going to change the check_cube_coordinates function please do it in a separate PR. It is used all over the place

Noted (I wasn't suggesting we do it here).

Reply:
@BelligerG Yes I agree you've put this in the right place now. It makes a lot more sense to go in the main blending plugin.

Reply:
I'm happy this is covered by a suitable test.